game_shared/ff/ff_utils.cpp

```diff
@@ -250,76 +250,57 @@ const char *FF_GetDefaultWeapon( const char *classname )
 {
-	Color innerCol;
-	if(!invertScale) 
+	Color innerCol = NULL;
+
+	if( iAmount <= iRed && iColorSetting > 0)
 	{
-		if( iAmount <= iRed && iColorSetting > 0)
-			innerCol = INTENSITYSCALE_COLOR_RED;
-		else if(iAmount  <= iOrange && iColorSetting > 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iRed, iOrange, INTENSITYSCALE_COLOR_RED, INTENSITYSCALE_COLOR_ORANGE );
-			else
-				innerCol = INTENSITYSCALE_COLOR_ORANGE;	
-		else if(iAmount  <= iYellow && iColorSetting> 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iOrange, iYellow, INTENSITYSCALE_COLOR_ORANGE, INTENSITYSCALE_COLOR_YELLOW );
-			else
-				innerCol = INTENSITYSCALE_COLOR_YELLOW;
-		else if(iColorSetting > 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iYellow, iGreen, INTENSITYSCALE_COLOR_YELLOW, INTENSITYSCALE_COLOR_GREEN );
-			else
-				innerCol = INTENSITYSCALE_COLOR_GREEN;
+		innerCol = INTENSITYSCALE_COLOR_RED;
+	}
+	else if(iAmount  <= iOrange && iColorSetting > 0)
+	{
+		if(iColorSetting == 2)
+			innerCol = ColorFade( iAmount, iRed, iOrange, INTENSITYSCALE_COLOR_RED, INTENSITYSCALE_COLOR_ORANGE );
 		else
-			innerCol = INTENSITYSCALE_COLOR_DEFAULT;
-
-		return *new Color(innerCol.r(), innerCol.g(), innerCol.b(), iAlpha);
+			innerCol = INTENSITYSCALE_COLOR_ORANGE;	
 	}
-	else
+	else if(iAmount  <= iYellow && iColorSetting> 0)
 	{
-		//not working
-		if( iAmount > iRed && iColorSetting > 0)
-			innerCol = INTENSITYSCALE_COLOR_RED;
-		else if(iAmount  > iOrange && iColorSetting > 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iOrange, iRed, INTENSITYSCALE_COLOR_ORANGE,INTENSITYSCALE_COLOR_RED );
-			else
-				innerCol = INTENSITYSCALE_COLOR_ORANGE;	
-		else if(iAmount  > iYellow && iColorSetting> 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iYellow, iOrange, INTENSITYSCALE_COLOR_YELLOW, INTENSITYSCALE_COLOR_ORANGE );
-			else
-				innerCol = INTENSITYSCALE_COLOR_YELLOW;
-		else if(iColorSetting > 0)
-			if(iColorSetting == 2)
-				innerCol = ColorFade( iAmount, iGreen, iYellow, INTENSITYSCALE_COLOR_YELLOW, INTENSITYSCALE_COLOR_GREEN );
-			else
-				innerCol = INTENSITYSCALE_COLOR_GREEN;
+		if(iColorSetting == 2)
+			innerCol = ColorFade( iAmount, iOrange, iYellow, INTENSITYSCALE_COLOR_ORANGE, INTENSITYSCALE_COLOR_YELLOW );
 		else
-			innerCol = INTENSITYSCALE_COLOR_DEFAULT;
-
-		return *new Color(innerCol.r(), innerCol.g(), innerCol.b(), iAlpha);
+			innerCol = INTENSITYSCALE_COLOR_YELLOW;
+	}
+	else if(iColorSetting > 0)
+	{
+		if(iColorSetting == 2)
+			innerCol = ColorFade( iAmount, iYellow, iGreen, INTENSITYSCALE_COLOR_YELLOW, INTENSITYSCALE_COLOR_GREEN );
+		else
+			innerCol = INTENSITYSCALE_COLOR_GREEN;
 	}
+	else
+	{
+		innerCol = INTENSITYSCALE_COLOR_DEFAULT;
+	}
+
+	return *new Color(innerCol.r(), innerCol.g(), innerCol.b(), iAlpha);

```

Comment:
Not specific to this PR since it was doing this previously as well, but won't this leak the Color? Why isn't this just returning a stack-allocated Color?

Reply:
I will address none the less if I figure out what it is that needs to change. Didn’t look too closely at the time. I’ll ask if I’m unsure

Reply:
Should be as simple as

```suggestion
	return Color(innerCol.r(), innerCol.g(), innerCol.b(), iAlpha);
```

Reply:
Cool, I’ve not done C++ since my last development 10years ago. So the oddities of it are lost on me after years of C# and JavaScript

Reply:
@squeek502 fixed in commit https://github.com/fortressforever/fortressforever/pull/381/commits/010e5a5af60ccef3c071d4f696893c8a2723cae3